File: OCH_Tao/ABC/B/368.js
const A = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")[1].split(" ").map(x => parseInt(x, 10));
let cnt = 0;
while (true) {
  cnt++;
  A.sort((a, b) => b - a);
  A[0]--;
  A[1]--;
  A.sort((a, b) => b - a);
  if (A[1] < 1) {
    console.log(cnt);
    break;
  }
}

File: OCH_Tao/ABC/B/368.py
N = int(input())
A = list(map(int,input().split()))
cnt = 0
while True:
  cnt+=1
  A.sort(reverse=True)
  A[0]-=1
  A[1]-=1
  if sorted(A,reverse=True)[1]<1:
    print(cnt)
    break